Issue/2566 order open event

AmandaRiu opened the pull request:
Closes #2566 The `ORDER_OPEN` track event was erroneously removed in [PR 1559](https://github.com/woocommerce/woocommerce-android/pull/1559) when the `OrderListPresenter` was deleted. All requests to open an order are routed through `MainActivity` so I've added the logic there to send the track event.


## Testing
1. From the Order List screen, tap an order. -> Check that in the console you are able to see the `order_open` event logged.  You should see something similar to this: `Tracked: order_open, Properties: {"id":9121,"status":"processing","blog_id":1111,"is_wpcom_store":false}`
2. No app changes should be introduced, please feel free to sanity check that you are able to open an order detail from all the entry points.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Fix lint check for assertion

Commit:
Send the OPEN_ORDER track events when opening order detail

anitaa1990 reviewed (dismissed):
Looks good! Tested by clicking an order from the order list, marking an order as fulfiled from Order Fulfill screen and by clicking on a new order notification. The event is sent in all 3 scenarios 🚢

anitaa1990 commented:
@AmandaRiu the code changes look good and I've approved the PR but CircleCI lint is blocking me from merging the PR. Let me know once it is fixed and we can merge. Thanks!

Commit:
Fix ktlint errors

Commit:
Only send the open order event when user is requesting it

This block of code ensures that the request to return to the order detail view
after marking an order complete does not trigger the OPEN-ORDER track event.

AmandaRiu commented:
@anitaa1990 I've fixed the ktlint errors and also added some additional logic to ensure that the `open-order` event is only fired when the user requests to see the order detail and _not_ when coming back from fulfilling an order. You can test this by opening an order with the status of "processing" and verifying the event is fired, and then marking the order complete and verifying no event is sent when the user returns to that same order detail screen.

AmandaRiu commented:
Taking this one back to tweak the logic since iOS doesn't send this event when opening an order from a notification.

AmandaRiu commented:
Instead of reverting all this logic I'm going to close this PR and open a new one with a clean branch.
